### PR TITLE
[Refactor] Update from single file

### PR DESCRIPTION
**Pull request by DN6**
# What does this PR do?
`from_single_file` relies on `download_from_original_stable_diffusion_ckpt` which is now quite difficult to read and has a number of subtle issues. 

This PR:

- Refactors from single file so that it now builds the individual components of the invoking pipeline class and creates the pipeline. This ensures that the correct pipeline class is always created when using from single file.  
- Removes circular dependencies in the original single file mixin by removing imports of the actual pipeline objects
- Removes multiple if-else blocks in favour of using dedicated functions to download the relevant configs and extract parameters from checkpoints. 
- Moves `FromOriginalVAEMixin` and `FromOriginalControlnetMixin` into their own modules. 

Some Notes:
- Tests for the single file mixin rely on using real checkpoints with `test_download_ckpt_diff_format_is_same`. As such, only slow tests have been added.  

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [ ] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ ] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/diffusers/loaders/autoencoder.py [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+    def from_single_file(cls, pretrained_model_link_or_path, **kwargs):
+        r"""
+        Instantiate a [`AutoencoderKL`] from pretrained ControlNet weights saved in the original `.ckpt` or
+        `.safetensors` format. The pipeline is set in evaluation mode (`model.eval()`) by default.
+
+        Parameters:
+            pretrained_model_link_or_path (`str` or `os.PathLike`, *optional*):
+                Can be either:
+                    - A link to the `.ckpt` file (for example
+                      `"https://huggingface.co/<repo_id>/blob/main/<path_to_file>.ckpt"`) on the Hub.
```
Comment (patrickvonplaten) [inline src/diffusers/loaders/autoencoder.py:44]:
```suggestion
                      `"https://huggingface.co/<repo_id>/blob/main/<path_to_file>.safetensors"`) on the Hub.
```

File: src/diffusers/loaders/autoencoder.py [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+                    - A path to a *file* containing all pipeline weights.
+            torch_dtype (`str` or `torch.dtype`, *optional*):
+                Override the default `torch.dtype` and load the model with another dtype. If `"auto"` is passed, the
```
Comment (patrickvonplaten) [inline src/diffusers/loaders/autoencoder.py:47]: We allow passing `"auto"` here?

File: src/diffusers/loaders/autoencoder.py [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+    is_accelerate_available,
+    is_omegaconf_available,
+    is_transformers_available,
+    logging,
+)
+from ..utils.import_utils import BACKENDS_MAPPING
+
+
+if is_transformers_available():
+    pass
```
Comment (patrickvonplaten) [inline src/diffusers/loaders/autoencoder.py:20]:
```suggestion
if is_transformers_available():
    pass
```

What is the purpose of this?

File: src/diffusers/loaders/autoencoder.py [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+            config_url = "https://raw.githubusercontent.com/CompVis/stable-diffusion/main/configs/stable-diffusion/v1-inference.yaml"
+            config_file = BytesIO(requests.get(config_url).content)
+
+        original_config = OmegaConf.load(config_file)
+
+        # default to sd-v1-5
+        image_size = image_size or 512
+
+        vae_config = create_vae_diffusers_config(original_config, image_size=image_size)
+        converted_vae_checkpoint = convert_ldm_vae_checkpoint(checkpoint, vae_config)
```
Comment (patrickvonplaten) [inline src/diffusers/loaders/autoencoder.py:195]: We should refactor the `convert_ldm_vae_checkpoint` function

File: src/diffusers/loaders/controlnet.py [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+                repo_id,
+                filename=file_path,
+                cache_dir=cache_dir,
+                resume_download=resume_download,
+                proxies=proxies,
+                local_files_only=local_files_only,
+                token=token,
+                revision=revision,
+                force_download=force_download,
+            )
```
Comment (patrickvonplaten) [inline src/diffusers/loaders/controlnet.py:145]:
```suggestion
        file_extension = pretrained_model_link_or_path.rsplit(".", 1)[-1]
        from_safetensors = file_extension == "safetensors"

        if from_safetensors and use_safetensors is False:
            raise ValueError("Make sure to install `safetensors` with `pip install safetensors`.")

        # remove huggingface url
        for prefix in ["https://huggingface.co/", "huggingface.co/", "hf.co/", "https://hf.co/"]:
            if pretrained_model_link_or_path.startswith(prefix):
                pretrained_model_link_or_path = pretrained_model_link_or_path[len(prefix) :]

        # Code based on diffusers.pipelines.pipeline_utils.DiffusionPipeline.from_pretrained
        ckpt_path = Path(pretrained_model_link_or_path)
        if not ckpt_path.is_file():
            # get repo_id and (potentially nested) file path of ckpt in repo
            repo_id = "/".join(ckpt_path.parts[:2])
            file_path = "/".join(ckpt_path.parts[2:])

            if file_path.startswith("blob/"):
                file_path = file_path[len("blob/") :]

            if file_path.startswith("main/"):
                file_path = file_path[len("main/") :]

            pretrained_model_link_or_path = hf_hub_download(
                repo_id,
                filename=file_path,
                cache_dir=cache_dir,
                resume_download=resume_download,
                proxies=proxies,
                local_files_only=local_files_only,
                token=token,
                revision=revision,
                force_download=force_download,
            )
```

This code is the same for all files, let's factor this into a function that is shared

File: src/diffusers/loaders/controlnet.py [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+
+        if config_file is None:
+            config_url = "https://raw.githubusercontent.com/lllyasviel/ControlNet/main/models/cldm_v15.yaml"
+            config_file = BytesIO(requests.get(config_url).content)
+
+        image_size = image_size or 512
+
+        controlnet = download_controlnet_from_original_ckpt(
```
Comment (patrickvonplaten) [inline src/diffusers/loaders/controlnet.py:153]: This function needs to be refactored

File: src/diffusers/loaders/single_file_utils.py [hunk 7]
```diff
@@ -0,0 +1,1547 @@
...
+
+        #         new_item = new_item.replace('norm.weight', 'group_norm.weight')
+        #         new_item = new_item.replace('norm.bias', 'group_norm.bias')
+
+        #         new_item = new_item.replace('proj_out.weight', 'proj_attn.weight')
+        #         new_item = new_item.replace('proj_out.bias', 'proj_attn.bias')
+
+        #         new_item = shave_segments(new_item, n_shave_prefix_segments=n_shave_prefix_segments)
+
+        mapping.append({"old": old_item, "new": new_item})
```
Comment (patrickvonplaten) [inline src/diffusers/loaders/single_file_utils.py:311]:
```suggestion

        #         new_item = new_item.replace('norm.weight', 'group_norm.weight')
        #         new_item = new_item.replace('norm.bias', 'group_norm.bias')

        #         new_item = new_item.replace('proj_out.weight', 'proj_attn.weight')
        #         new_item = new_item.replace('proj_out.bias', 'proj_attn.bias')

        #         new_item = shave_segments(new_item, n_shave_prefix_segments=n_shave_prefix_segments)

        mapping.append({"old": old_item, "new": new_item})
```
Let's remove the comments if they are not needed

File: src/diffusers/loaders/single_file_utils.py [hunk 8]
```diff
@@ -0,0 +1,1547 @@
...
+
+        new_item = shave_segments(new_item, n_shave_prefix_segments=n_shave_prefix_segments)
+
+        mapping.append({"old": old_item, "new": new_item})
+
+    return mapping
+
+
+# Copied from diffusers.pipelines.stable_diffusion.convert_from_ckpt.assign_to_checkpoint
+def assign_to_checkpoint(
```
Comment (patrickvonplaten) [inline src/diffusers/loaders/single_file_utils.py:348]: Let's try to make the whole conversion process more robust instead of just copying the functions here. We should not need 1000 lines to convert the checkpoint. Let's try to just have one big mapping that can be applied to all models

File: src/diffusers/loaders/single_file_utils.py [hunk 9]
```diff
@@ -0,0 +1,1547 @@
...
+            new_checkpoint[f"controlnet_cond_embedding.blocks.{diffusers_index}.weight"] = unet_state_dict.pop(
+                f"input_hint_block.{orig_index}.weight"
+            )
+            new_checkpoint[f"controlnet_cond_embedding.blocks.{diffusers_index}.bias"] = unet_state_dict.pop(
+                f"input_hint_block.{orig_index}.bias"
+            )
+            diffusers_index += 1
+            orig_index += 2
+
+        new_checkpoint["controlnet_cond_embedding.conv_out.weight"] = unet_state_dict.pop(
```
Comment (patrickvonplaten) [inline src/diffusers/loaders/single_file_utils.py:776]:
```suggestion
        orig_index = 0

        new_checkpoint["controlnet_cond_embedding.conv_in.weight"] = unet_state_dict.pop(
            f"input_hint_block.{orig_index}.weight"
        )
        new_checkpoint["controlnet_cond_embedding.conv_in.bias"] = unet_state_dict.pop(
            f"input_hint_block.{orig_index}.bias"
        )

        orig_index += 2

        diffusers_index = 0

        while diffusers_index < 6:
            new_checkpoint[f"controlnet_cond_embedding.blocks.{diffusers_index}.weight"] = unet_state_dict.pop(
                f"input_hint_block.{orig_index}.weight"
            )
            new_checkpoint[f"controlnet_cond_embedding.blocks.{diffusers_index}.bias"] = unet_state_dict.pop(
                f"input_hint_block.{orig_index}.bias"
            )
            diffusers_index += 1
            orig_index += 2

        new_checkpoint["controlnet_cond_embedding.conv_out.weight"] = unet_state_dict.pop(
```

This is very hacky - there has to be a better, more robust way to convert the checkpoint

File: src/diffusers/loaders/single_file_utils.py [hunk 10]
```diff
@@ -0,0 +1,1547 @@
...
+    new_checkpoint["decoder.conv_out.weight"] = vae_state_dict["decoder.conv_out.weight"]
+    new_checkpoint["decoder.conv_out.bias"] = vae_state_dict["decoder.conv_out.bias"]
+    new_checkpoint["decoder.conv_norm_out.weight"] = vae_state_dict["decoder.norm_out.weight"]
+    new_checkpoint["decoder.conv_norm_out.bias"] = vae_state_dict["decoder.norm_out.bias"]
+
+    new_checkpoint["quant_conv.weight"] = vae_state_dict["quant_conv.weight"]
+    new_checkpoint["quant_conv.bias"] = vae_state_dict["quant_conv.bias"]
+    new_checkpoint["post_quant_conv.weight"] = vae_state_dict["post_quant_conv.weight"]
+    new_checkpoint["post_quant_conv.bias"] = vae_state_dict["post_quant_conv.bias"]
+
```
Comment (patrickvonplaten) [inline src/diffusers/loaders/single_file_utils.py:825]:
```suggestion
    new_checkpoint["encoder.conv_in.weight"] = vae_state_dict["encoder.conv_in.weight"]
    new_checkpoint["encoder.conv_in.bias"] = vae_state_dict["encoder.conv_in.bias"]
    new_checkpoint["encoder.conv_out.weight"] = vae_state_dict["encoder.conv_out.weight"]
    new_checkpoint["encoder.conv_out.bias"] = vae_state_dict["encoder.conv_out.bias"]
    new_checkpoint["encoder.conv_norm_out.weight"] = vae_state_dict["encoder.norm_out.weight"]
    new_checkpoint["encoder.conv_norm_out.bias"] = vae_state_dict["encoder.norm_out.bias"]

    new_checkpoint["decoder.conv_in.weight"] = vae_state_dict["decoder.conv_in.weight"]
    new_checkpoint["decoder.conv_in.bias"] = vae_state_dict["decoder.conv_in.bias"]
    new_checkpoint["decoder.conv_out.weight"] = vae_state_dict["decoder.conv_out.weight"]
    new_checkpoint["decoder.conv_out.bias"] = vae_state_dict["decoder.conv_out.bias"]
    new_checkpoint["decoder.conv_norm_out.weight"] = vae_state_dict["decoder.norm_out.weight"]
    new_checkpoint["decoder.conv_norm_out.bias"] = vae_state_dict["decoder.norm_out.bias"]

    new_checkpoint["quant_conv.weight"] = vae_state_dict["quant_conv.weight"]
    new_checkpoint["quant_conv.bias"] = vae_state_dict["quant_conv.bias"]
    new_checkpoint["post_quant_conv.weight"] = vae_state_dict["post_quant_conv.weight"]
    new_checkpoint["post_quant_conv.bias"] = vae_state_dict["post_quant_conv.bias"]

```

This can be better solved with a big mapping

File: src/diffusers/loaders/autoencoder.py (hunk 1 above)
Comment (sayakpaul) [reply src/diffusers/loaders/autoencoder.py:44]: Is `.ckpt` still used? If we know that to be "yes" I think it would make sense to add a comment below that ".ckpt" files are also supported.

File: src/diffusers/loaders/autoencoder.py (hunk 4 above)
Comment (sayakpaul) [reply src/diffusers/loaders/autoencoder.py:195]: Could it make sense to have in a followup PR?

File: src/diffusers/loaders/controlnet.py [hunk 11]
```diff
@@ -0,0 +1,167 @@
...
+        Examples:
+
+        ```py
+        from diffusers import StableDiffusionControlNetPipeline, ControlNetModel
+
+        url = "https://huggingface.co/lllyasviel/ControlNet-v1-1/blob/main/control_v11p_sd15_canny.pth"  # can also be a local path
+        model = ControlNetModel.from_single_file(url)
+
+        url = "https://huggingface.co/runwayml/stable-diffusion-v1-5/blob/main/v1-5-pruned.safetensors"  # can also be a local path
+        pipe = StableDiffusionControlNetPipeline.from_single_file(url, controlnet=controlnet)
```
Comment (sayakpaul) [inline src/diffusers/loaders/controlnet.py:87]: For housekeeping, let's make sure we have these examples (working examples, of course) for all the single file loader mixins.

File: src/diffusers/loaders/controlnet.py (hunk 5 above)
Comment (sayakpaul) [reply src/diffusers/loaders/controlnet.py:145]: Maybe put it in a utility script called `single_file_utils.py`.

File: src/diffusers/loaders/single_file.py [hunk 12]
```diff
@@ -11,45 +11,202 @@
...
+    infer_model_type,
+)
 
 
 if is_transformers_available():
     pass
 
 if is_accelerate_available():
-    from accelerate import init_empty_weights
+    pass
```
Comment (sayakpaul) [inline src/diffusers/loaders/single_file.py:43]: Why is this needed?

**Review by sayakpaul — COMMENTED** [review]
Nice start! 

I think we can still go for fast tests. For that, we could use the tiny pipelines we have for testing, and convert to the single checkpoint format. This ensures a quick and fast way to check if the utilities are working as expected. But this requires configuring the state dict mappings too much, I am okay with things being as is. 

Another point would be to make sure to update the docs for the newly added and separated loader classes.

File: src/diffusers/loaders/single_file.py [hunk 13]
```diff
@@ -11,39 +11,154 @@
...
-
-import requests
-import torch
-import yaml
-from huggingface_hub import hf_hub_download
+import os
+import re
+
 from huggingface_hub.utils import validate_hf_hub_args
+from transformers import AutoFeatureExtractor
```
Comment (patrickvonplaten) [inline src/diffusers/loaders/single_file.py:18]:
```suggestion
if is_transformers_available():
    from transformers import AutoFeatureExtractor
```

File: src/diffusers/loaders/single_file.py [hunk 14]
```diff
@@ -11,39 +11,154 @@
...
-from ..utils import deprecate, is_accelerate_available, is_transformers_available, logging
 
+logger = logging.get_logger(__name__)
 
-if is_transformers_available():
-    pass
 
-if is_accelerate_available():
-    from accelerate import init_empty_weights
+VALID_URL_PREFIXES = ["https://huggingface.co/", "huggingface.co/", "hf.co/", "https://hf.co/"]
```
Comment (patrickvonplaten) [inline src/diffusers/loaders/single_file.py:39]:
```suggestion
VALID_HF_URL_PREFIXES = ["https://huggingface.co/", "huggingface.co/", "hf.co/", "https://hf.co/"]
```

File: src/diffusers/loaders/single_file.py [hunk 15]
```diff
@@ -11,39 +11,154 @@
...
+def build_sub_model_components(
+    pipeline_components,
+    pipeline_class_name,
+    component_name,
+    original_config,
+    checkpoint,
+    checkpoint_path_or_dict,
+    local_files_only=False,
+    load_safety_checker=False,
+    **kwargs,
```
Comment (patrickvonplaten) [inline src/diffusers/loaders/single_file.py:73]: Not a big fan of having `kwargs` in functions in general, but ok!

File: src/diffusers/loaders/single_file.py [hunk 16]
```diff
@@ -11,39 +11,154 @@
...
+        return {"safety_checker": safety_checker, "feature_extractor": feature_extractor}
+
+    return
+
+
+def set_additional_components(
+    pipeline_class_name,
+    original_config,
+    **kwargs,
+):
```
Comment (patrickvonplaten) [inline src/diffusers/loaders/single_file.py:141]:
```suggestion
def set_additional_components(pipeline_class_name, original_config, model_type=None):
```

File: src/diffusers/loaders/single_file.py [hunk 17]
```diff
@@ -11,39 +11,154 @@
...
+    components = {}
+    model_type = kwargs.get("model_type", None)
```
Comment (patrickvonplaten) [inline src/diffusers/loaders/single_file.py:143]:
```suggestion
```

File: src/diffusers/loaders/single_file.py [hunk 18]
```diff
@@ -11,39 +11,154 @@
...
+
+    return components
 
 
 class FromSingleFileMixin:
     """
     Load model weights saved in the `.ckpt` format into a [`DiffusionPipeline`].
     """
 
-    @classmethod
```
Comment (patrickvonplaten) [inline src/diffusers/loaders/single_file.py:41]: Nice!

File: src/diffusers/loaders/single_file.py [hunk 19]
```diff
@@ -143,484 +221,101 @@ def from_single_file(cls, pretrained_model_link_or_path, **kwargs):
...
-        extract_ema = kwargs.pop("extract_ema", False)
-        image_size = kwargs.pop("image_size", None)
-        scheduler_type = kwargs.pop("scheduler_type", "pndm")
-        num_in_channels = kwargs.pop("num_in_channels", None)
-        upcast_attention = kwargs.pop("upcast_attention", None)
-        load_safety_checker = kwargs.pop("load_safety_checker", True)
-        prediction_type = kwargs.pop("prediction_type", None)
-        text_encoder = kwargs.pop("text_encoder", None)
-        text_encoder_2 = kwargs.pop("text_encoder_2", None)
-        vae = kwargs.pop("vae", None)
```
Comment (patrickvonplaten) [inline src/diffusers/loaders/single_file.py:167]: We should still be able to manually pass the `vae` and `text_encoder` I think no?

File: src/diffusers/loaders/single_file.py [hunk 20]
```diff
@@ -143,484 +221,101 @@ def from_single_file(cls, pretrained_model_link_or_path, **kwargs):
...
-        use_linear_projection = kwargs.pop("use_linear_projection", None)
-        revision = kwargs.pop("revision", None)
-        extract_ema = kwargs.pop("extract_ema", False)
-        image_size = kwargs.pop("image_size", None)
-        upcast_attention = kwargs.pop("upcast_attention", None)
-
-        torch_dtype = kwargs.pop("torch_dtype", None)
-
-        use_safetensors = kwargs.pop("use_safetensors", None)
+        if class_name == "AutoencoderKL":
```
Comment (patrickvonplaten) [inline src/diffusers/loaders/single_file.py:265]:
```suggestion
        vae = kwargs.pop("vae", None)
        if class_name == "AutoencoderKL" and vae is None:
```

File: src/diffusers/pipelines/pipeline_utils.py [hunk 21]
```diff
@@ -389,7 +389,12 @@ def _get_pipeline_class(
         return class_obj
 
     diffusers_module = importlib.import_module(class_obj.__module__.split(".")[0])
-    class_name = config["_class_name"]
+    class_name = class_name or config["_class_name"]
```
Comment (patrickvonplaten) [inline src/diffusers/pipelines/pipeline_utils.py:392]: Is that fully backwards compatible?

File: src/diffusers/loaders/single_file.py [hunk 22]
```diff
@@ -143,484 +221,101 @@ def from_single_file(cls, pretrained_model_link_or_path, **kwargs):
...
+            image_size = kwargs.pop("image_size", None)
+            component = create_diffusers_vae_model_from_ldm(
+                class_name, original_config, checkpoint, image_size=image_size
+            )
+            return component["vae"]
```
Comment (patrickvonplaten) [inline src/diffusers/loaders/single_file.py:270]: This function can return both a pipeline and a model. It's good that we re-use the same loading logic here, but I still wonder if we should maybe make have two different "single file" loading classes - one for pipelines and one for models. The one for models can be much shorter. Wdyt?

**Comment by patrickvonplaten** [comment]
Nice PR - things look more or less good to go for me. Some things I'd like to resolve before merging:

- 1.) Do we want to re-use the same "single-file" loading class for both models and pipelines (see comment [here](https://github.com/huggingface/diffusers/pull/6428#discussion_r1458665094)). Could it maybe be better to write a shorter `FromSingleModelFileMixin` class for models? 
- 2.) When changing code in `pipeline_utils.py` we need to be extra careful. Let's make sure to monitor all slow tests once this PR is merged
- 3.) We should add some fast tests here as well. Can you take some of our `diffusers-format` tiny checkpoints: https://huggingface.co/hf-internal-testing/tiny-stable-diffusion-pipe

File: src/diffusers/loaders/single_file.py (hunk 22 above)
Comment (DN6) [reply src/diffusers/loaders/single_file.py]: Yeah I think so too. I've updated.

File: src/diffusers/models/autoencoders/autoencoder_kl_temporal_decoder.py [hunk 23]
```diff
@@ -162,7 +161,7 @@ def custom_forward(*inputs):
         return sample
 
 
-class AutoencoderKLTemporalDecoder(ModelMixin, ConfigMixin, FromOriginalVAEMixin):
+class AutoencoderKLTemporalDecoder(ModelMixin, ConfigMixin):
```
[the remark this answers is not in the document]
Comment (DN6) [reply src/diffusers/models/autoencoders/autoencoder_kl_temporal_decoder.py:164]: Yeah I think this was just copied over from `AutoencoderKL`.

File: src/diffusers/loaders/single_file.py (hunk 19 above)
Comment (DN6) [reply src/diffusers/loaders/single_file.py:167]: It's parsed from the kwargs similar to `from_pretrained`
https://github.com/huggingface/diffusers/blob/36203576d5c8e58b0dc823207c6628599d70a343/src/diffusers/loaders/single_file.py#L234-L236

**Review by patrickvonplaten — APPROVED** [review]
PR is good to merge for me to get going here. Let's make sure we also refactor the "model-single-file" loaders in a follow-up PR.

**Comment by DN6** [comment]
I checked and the the slow tests are passing. I can add the model `single-file` refactor and fast tests in a follow up.

File: src/diffusers/pipelines/pipeline_utils.py (hunk 21 above)
Comment (DN6) [reply src/diffusers/pipelines/pipeline_utils.py:392]: I searched the code base and it seems like `class_name` or config is always passed here.

File: src/diffusers/loaders/single_file.py [hunk 24]
```diff
@@ -143,484 +199,80 @@ def from_single_file(cls, pretrained_model_link_or_path, **kwargs):
...
-
-        if scaling_factor is None:
-            if (
-                "model" in original_config
-                and "params" in original_config["model"]
-                and "scale_factor" in original_config["model"]["params"]
-            ):
-                vae_scaling_factor = original_config["model"]["params"]["scale_factor"]
-            else:
-                vae_scaling_factor = 0.18215  # default SD scaling factor
```
Comment (patrickvonplaten) [inline src/diffusers/loaders/single_file.py:460]: If no scaling factor is provided, the default should be set to 0.18215
